Mathlib/NumberTheory/LSeries/SumCoeff.lean

```diff
@@ -153,3 +160,222 @@ theorem LSeries_eq_mul_integral_of_nonneg (f : ℕ → ℝ) {r : ℝ} (hr : 0 
   LSeries_eq_mul_integral' _ hr hs <| hO.congr_left fun _ ↦ by simp [_root_.abs_of_nonneg (hf _)]
 
 end integralrepresentation
+
+noncomputable section residue
+
+variable {f : ℕ → ℂ} {l : ℂ} (hlim : Tendsto (fun n : ℕ ↦ (∑ k ∈ Icc 1 n, f k) / n) atTop (𝓝 l))
+
+section lemmas
+
+include hlim in
+private theorem lemma₁ {s : ℝ} (hs : 1 < s) :
+    IntegrableOn (fun t : ℝ ↦ (∑ k ∈ Icc 1 ⌊t⌋₊, f k) * (t : ℂ) ^ (-(s : ℂ) - 1)) (Set.Ici 1) := by
+  have h₁ : LocallyIntegrableOn (fun t : ℝ ↦ (∑ k ∈ Icc 1 ⌊t⌋₊, f k) * (t : ℂ) ^ (-(s : ℂ) - 1))
+        (Set.Ici 1) := by
+    simp_rw [mul_comm]
+    refine locallyIntegrableOn_mul_sum_Icc f zero_le_one ?_
+    refine ContinuousOn.locallyIntegrableOn (fun t ht ↦ ?_) measurableSet_Ici
+    exact (continuousAt_ofReal_cpow_const _ _ <|
+      Or.inr (zero_lt_one.trans_le ht).ne').continuousWithinAt
+  have h₂ : (fun t : ℝ ↦ ∑ k ∈ Icc 1 ⌊t⌋₊, f k) =O[atTop] fun t ↦ t ^ (1 : ℝ) := by
+    simp_rw [Real.rpow_one]
+    refine IsBigO.trans_isEquivalent ?_ isEquivalent_nat_floor
+    suffices Tendsto (fun n ↦ (∑ k ∈ Icc 1 n, f k) / ((n : ℝ) ^ (1 : ℝ) : ℝ)) atTop (𝓝 l) by
```

Comment:
The body of the `suffices` is shorter than the remaining proof, so maybe use `have`?